[SC-356] Usprawnienie działania json editora

dmncp opened the pull request:
W tym tasku:
1. Opcja pobrania zawartości wyświetlanej w json editorze (w pliku json)
2. Opcja uploadu pliku json zamiast wysyłania z json editora

Powyższe dotyczy zarówno edycji jak i dodawania aktywności.

dmncp committed:
[SC-356] feature: download and upload json during activity edition

dmncp committed:
[SC-356] feature: file handling for activity edition and addition

kwdrt commented on frontend/src/utils/constants.js:
```diff
@@ -269,3 +269,10 @@ export const BadgeType = {
+  [Activity.INFO]: 'info-task-configuration.json',
+  [Activity.TASK]: 'combat-task-configuration.json',
+  [Activity.SURVEY]: 'survey-task-configuration.json',
+  [Activity.EXPEDITION]: 'graph-task-configuration.json'
```
Można by się zastanowić czy nie chcemy robić nazwy pobieranego pliku jako <nazwa aktywności>-<rodzaj aktywności>, bo teraz przy pobieraniu będziemy mieć wiele plików o tej samej nazwie

dmncp replied:
No to problem prowadzącego + może sobie zmienić przy zapisie. Ciężko by było dać nazwę aktywności w nazwie pliku przy dodawaniu nowej aktywności bo wtedy by i tak byłby domyślny string z json editora co by nie zlikwidowało tego problemu.

kwdrt reviewed (approved):
LGTM 👍 

Widzę jedno potencjalne usprawnienie, ale działa prawidłowo

dmncp committed:
[SC-356] Merge remote-tracking branch 'origin/develop' into SC-356-usprawnienie-dzialania-json-editora

# Conflicts:
#	frontend/src/components/professor/ChapterDetails/AddActivity/AddActivity.js
#	frontend/src/components/professor/ChapterDetails/AddActivity/AddGraphTask.js
#	frontend/src/components/professor/ChapterDetails/AddActivity/AddInfoTask.js
#	frontend/src/components/professor/ChapterDetails/EditActivityModal.js
#	frontend/src/utils/constants.js

dmncp committed:
[SC-356] refactor: remove unused imports
